Lib/test/test_call.py

```diff
@@ -11,6 +11,19 @@
 import contextlib
 
 
+class BadStr(str):
+    def __eq__(self, other):
+        return True
+    def __hash__(self):
+        # Guaranteed different hash
+        return str.__hash__(self) ^ 3
+
+    def __eq__(self, other):
+        return False
+    def __hash__(self):
+        return str.__hash__(self)
```

Comment:
This looks a little odd; aren't these latter `__eq__` and `__hash__` definitions rendering the first ones pointless?  Which set was supposed to be in this class?

Reply:
Good catch. The first definitions are correct, they cause a crash with non-patched code. The latter definitions do not cause any visible effects, Thank you, Zachary.